transport/http/inbound.go

```diff
@@ -51,6 +51,15 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 	}
 }
 
+// Interceptor specifies a func that takes the transport http.Handler and
+// produces an http.Handler. This can be used to create custom behavior
+// when inbound HTTP requests are recieved.
+func Interceptor(interceptor func(http.Handler) http.Handler) InboundOption {
```

Comment:
For documentation purposes, I would name the handler in the function.

    interceptor func(yarpcHandler http.Handler) http.Handler